deps/shui/src/logseq/shui/table/v2.cljs

```diff
@@ -0,0 +1,462 @@
+
+(defn cell-classes [table-opts]
+  (str/join " "
+    [(cell-bg-classes table-opts)
+     (cell-rounded-classes table-opts)
+     (cell-text-classes table-opts)
+     (cell-text-transform-classes table-opts)
+     (cell-padding-classes table-opts)]))
+
+;; -- Handlers -----------------------------------------------------------------
+
+(defn handle-cell-pointer-down [e {:keys [cell-focus col-index row-index]}] 
+  (when (not= cell-focus [col-index row-index])
+    (.stopPropagation e) 
+    (.preventDefault e)))
+
+(defn handle-cell-click 
+  "When a cell is clicked, we need to update the cursor position and the selected cells"
+  [e {:keys [cell-focus set-cell-focus header? col-index row-index]} cell-ref]
+  ; (.stopPropagation e) 
+  (.preventDefault e)
+  (when-not (= cell-focus [col-index row-index]) 
+    (set-cell-focus [col-index row-index])))
+    
+
+(defn handle-cell-keydown 
+  "When a cell is focused, we need to update the cursor position and the selected cells"
+  [e {:keys [cell-focus set-cell-focus header? col-index row-index total-rows total-cols]}]
+  (when (= cell-focus [col-index row-index]) 
+    (and (case (.-key e)
+           "ArrowUp"    (if (= row-index 0)
+                          (set-cell-focus [col-index row-index])
+                          (set-cell-focus [col-index (dec row-index)]))
+           "ArrowDown"  (if (= row-index (dec total-rows)) 
+                          (set-cell-focus [col-index row-index])
+                          (set-cell-focus [col-index (inc row-index)]))
+           "ArrowLeft"  (cond 
+                          ;; if we are in the top left, then do not move the focus
+                          (and (= col-index 0) (= row-index 0))
+                          (set-cell-focus [col-index row-index])
+                          ;; if we are in the first column, then move to the last column of the previous row
+                          (= col-index 0) 
+                          (set-cell-focus [(dec total-cols) (dec row-index)])
+                          ;; otherwise, move to the previous column
+                          :else
+                          (set-cell-focus [(dec col-index) row-index]))
+           "ArrowRight" (cond 
+                          ;; if we are in the bottom right, then do not move the focus
+                          (and (= col-index (dec total-cols)) (= row-index (dec total-rows))) 
+                          (set-cell-focus [col-index row-index])
+                          ;; if we are in the last column, then move to the first column of the next row
+                          (= col-index (dec total-cols)) 
+                          (set-cell-focus [0 (inc row-index)])
+                          ;; otherwise, move to the next column
+                          :else
+                          (set-cell-focus [(inc col-index) row-index]))
+           nil)
+         ;; Prevent default actions when the table handles it itself
+         (.preventDefault e))))
```

Comment:
One minor bug:
Press up/down/left/right will select or open a relative block. Should we disable the global shortcuts here(disable-all-shortcuts)? `stopPropagation` could be helpful too.

Reply:
I think the keyboard interactions need some love. We should probably disable all shortcuts when a cell is focused, but at the moment that would mean introducing frontend modules into shui, which we're trying to keep abstracted.

I think this is something that should be fixed before making v2 the default, but I think it is too much to fix in this initial PR. If it's a major sticking point for you I can fix it first thing after merge.